src/libsync/clientsideencryption.cpp

```diff
@@ -721,41 +721,41 @@ QByteArray encryptStringAsymmetric(EVP_PKEY *publicKey, const QByteArray& data)
     auto ctx = PKeyCtx::forKey(publicKey, ENGINE_get_default_RSA());
     if (!ctx) {
         qCInfo(lcCse()) << "Could not initialize the pkey context.";
-        exit(1);
```

Comment:
Callers are not going to handle that change in behavior, so likely needs some error handling up the call stack.

Reply:
Sadly there seems to be no actual error handling in `src/libsync/clientsideencryption.cpp`.
So for now the key is just skipped in `FolderMetadata::encryptedMetadata()`.

Reply:
Yes, definitely. This needs being addressed otherwise it's too much of a behavior change, this will start trickling up and not being blocked later on while right now it has no risk of trickling up and corrupting things (even though it does so in a super aggressive and less than ideal way indeed).

Reply:
Updated the error handling in the callers.